### PR TITLE
Fix error deleting attachments when deleting notes

### LINKED ISSUE #312: delete note broken since 3.3.0
seems code still contains references to `iat_status` column even it was dropped in #254.

```
[2017-10-10 15:26:48] app.ERROR: Uncaught Exception
Eventum\Db\DatabaseException: "SQLSTATE[42S22]: Column not found: 1054
Unknown column 'iat_status' in 'where clause'" at
/usr/share/eventum/src/Db/Adapter/PdoAdapter.php line 159
{"exception":"[object] (Eventum\\Db\\DatabaseException(code: 42S22):
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'iat_status' in
'where clause' at
/usr/share/eventum/src/Db/Adapter/PdoAdapter.php:159,
PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054
Unknown column 'iat_status' in 'where clause' at
/usr/share/eventum/src/Db/Adapter/PdoAdapter.php:157)"}
{"url":"/popup.php?cat=delete_note&id=169028"}
```

/cc @balsdorf

### PR DESCRIPTION
**Pull request by balsdorf**
Fixes #312

### CODE REVIEW

**Comment by glensc** [comment]
please rebase against master, your changelog entry is placed under wrong version. git rebase should fix that automatically, but check it before pushing over.

File: src/Attachment/AttachmentManager.php [hunk 1]
```diff
@@ -247,13 +247,19 @@ public static function cleanupAbandonedFiles()
      * issue in the database.
      *
      * @param   int $issue_id The issue ID
-     * @return  array The full list of attachments
+     * @param   int|null $min_role Don't show attachments with a role greater then this
```
Comment (glensc) [inline src/Attachment/AttachmentManager.php]: "show" is wrong term used in this context

File: src/Attachment/AttachmentManager.php [hunk 2]
```diff
@@ -269,14 +275,18 @@ public static function getList($issue_id)
...
         $stmt .= '
                  ORDER BY
                     iat_created_date ASC';
-        $params = [$issue_id, User::getRoleByUser($usr_id, $prj_id)];
         try {
             $res = DB_Helper::getInstance()->getAll($stmt, $params);
         } catch (DatabaseException $e) {
-            return '';
+            return [];
         }
```
Comment (glensc) [inline src/Attachment/AttachmentManager.php]: remove the try-catch. lets stop hiding errors.

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ v3.3.1 fixes packaging bug present in 3.3.0 version.
 
 - Include league flysystem traits in dist tarball (@glensc, #305)
 - RPC: check that user has access to issue (@glensc, #306)
+- Fix error deleting attachments when deleting notes (@balsdorf, #312)
```
Comment (glensc) [inline CHANGELOG.md]: include this PR id as well

Comment (balsdorf) [reply CHANGELOG.md]: Done

**Pull request merged**